diff --git a/src/PhpPresentation/Shape/AbstractDrawing.php b/src/PhpPresentation/Shape/AbstractDrawing.php
index 1a11119..65e7a4a 100644
--- a/src/PhpPresentation/Shape/AbstractDrawing.php
+++ b/src/PhpPresentation/Shape/AbstractDrawing.php
@@ -30,14 +30,14 @@ abstract class AbstractDrawing extends AbstractShape implements ComparableInterf
      *
      * @var int
      */
-    private static $imageCounter = 0;
+    protected static $imageCounter = 0;
 
     /**
      * Image index
      *
      * @var int
      */
-    private $imageIndex = 0;
+    protected $imageIndex = 0;
 
     /**
      * Name
diff --git a/src/PhpPresentation/Shape/Drawing.php b/src/PhpPresentation/Shape/Drawing.php
index 91651bf..3fb49dd 100644
--- a/src/PhpPresentation/Shape/Drawing.php
+++ b/src/PhpPresentation/Shape/Drawing.php
@@ -32,6 +32,13 @@ class Drawing extends AbstractDrawing implements ComparableInterface
     private $path;
 
     /**
+     * Cache
+     *
+     * @static var array
+     */
+    private static $indexCache = array();
+
+    /**
      * Create a new \PhpOffice\PhpPresentation\Slide\Drawing
      */
     public function __construct()
@@ -110,6 +117,11 @@ class Drawing extends AbstractDrawing implements ComparableInterface
             $this->path = $pValue;
         }
 
+        if (isset(self::$indexCache[$this->path])) {
+            $this->imageIndex = self::$indexCache[$this->path];
+        } else {
+            self::$indexCache[$this->path] = $this->imageIndex;
+        }
         return $this;
     }
 
